src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,324 @@
+                            Deadline deadline = new Deadline(description.substring(0, description.indexOf("(")), by);
+                            deadline.description = deadline.description.replaceFirst(" ", "X");
+                            deadline.isDone = true;
+                            doneList.add(deadline);
+                            continue;
+                        case "E":
+                            String at = tasks[i].substring(tasks[i].indexOf(":") + 2, tasks[i].indexOf(")"));
+                            Event event = new Event(description.substring(0, description.indexOf("(")), at);
+                            event.description = event.description.replaceFirst(" ", "X");
+                            event.isDone = true;
+                            doneList.add(event);
+                            continue;
+                        default:
+                            System.out.println("There are errors in reading in files.");
+                        }
+                    }
+                }
+                printDoneList();
+            } catch (StringIndexOutOfBoundsException e) {
+                printDoneList();
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("No files saved. Please start using Duke now!");
+            System.out.println(SPLIT_LINE);
+        }
+    }
+    public static void terminateProgram() {
+        System.out.println(SPLIT_LINE);
+        System.out.println("    Bye. Hope to see you again soon!\n" + SPLIT_LINE);
+    }
+
+    public static void printList() {
+        if (list.size() == 0) {
+            System.out.println(SPLIT_LINE);
+            System.out.println("    There is no task in your list!!");
+        } else {
+            System.out.println(SPLIT_LINE + "\n    Here are the tasks in your list:");
+            for (int i = 0; i < list.size(); i++) {
+                System.out.println("    " + (i + 1) + "." + list.get(i));
+            }
+        }
+        System.out.println(SPLIT_LINE);
+    }
+
+    public static void printDoneList() {
+        if (doneList.size() == 0) {
+            System.out.println(SPLIT_LINE);
+            System.out.println("    You have not finished any tasks");
+        } else {
+            System.out.println("    Congratulations! Now you have done " + doneList.size() + " task(s):");
+            for (int i = 0; i < doneList.size(); i++) {
+                System.out.println("    " + (i + 1) + "." + doneList.get(i));
+            }
+        }
+        System.out.println(SPLIT_LINE);
+    }
+
+    public static void doneTask(String task) {
+        task = task.trim();
+        String str = "";
+        if (task != null && !"".equals(task)) {
+            for (int i = 0; i < task.length(); i++) {
+                if (task.charAt(i) >= 48 && task.charAt(i) <= 57) {
+                    str = str + task.charAt(i);
+                }
+            }
```

Comment:
Having so many nested if statements and loops can be hard to read. Consider abstracting it out into a different function